wandb/sdk/internal/handler.py

```diff
@@ -168,9 +170,26 @@ def _save_history(self, record: Record) -> None:
                 self._sampled_history[k].add(v)
 
     def handle_history(self, record: Record) -> None:
+        history_dict = proto_util.dict_from_proto_list(record.history.item)
+
+        try:
+            has_step = record.step
+        except Exception:
```

Comment:
Catching exception here is overkill.  It would be an attribute error, but I think we'll always have the step attribute it would just be None...